src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py

```diff
@@ -287,6 +302,35 @@ def receive_trial_result(self, parameter_id, parameters, value, **kwargs):
             raise RuntimeError('Received parameter_id not in total_data.')
         params = self.total_data[parameter_id]
 
+        # code for parallel
+        if self.parallel:
+            constant_liar = False
+            if 'constant_liar' in kwargs:
+                constant_liar = kwargs['constant_liar']
+
+            if constant_liar:
+                rval = self.CL_rval
+            else:
+                rval = self.rval
```

Comment:
what is the meaning of line 312 and 314?

Reply:
@PurityFan

Reply:
since line 293 is removed, no more question here.